<!-- Author: Karla Stepanova -->
<launch>
    <arg name="bag" default="/home/tradr/Downloads/extracted_for_Karla/2017-08-07-14-54-52-recomputed_with_cameras_shorter.bag"/>
    <arg name="start" default="50.0"/>
    <arg name="duration" default="1000000.0" />
    <arg name="rate"  default="1.0"/>
    <arg name="delay" default="0.0"/>
    <arg name="args" default="-k"/>
    <arg name="remove_static_tf" default="false"/>
    <arg name="load_params" default="false" />
    <arg name="use_old_robot" default="false"/>
    <arg name="required" default="false" />
    <arg name="other_args" default="" />
    <arg name="cam_used" default="camera_4" />
    <arg name="cam_dest" default="viz" />
    <arg name="img_dest" default="image/compressed" />

    <param name="/use_sim_time" value="true"/>
    <param name="camera_used" value="$(arg cam_used)" />
    <param name="image_spec" value="$(arg img_dest)" />
    <param name="camera_dest" value ="$(arg cam_dest)" />
    <rosparam>
        used_classes: [0,1,4,5,6,9,10]
        sel_classes: [1 2]
        inputImgWidth: 1616
        inputImgHeight: 1232
    </rosparam>

    <node name="rosbag_play" pkg="rosbag" type="play" args="--clock -d $(arg delay) -s $(arg start) -u $(arg duration) -r $(arg rate) $(arg args) $(arg other_args) --bags $(arg bag)" output="screen" required="$(arg required)">
      <remap if="$(arg remove_static_tf)" from="tf" to="tf_old"/>
      <remap from="rosout" to="null" />
      <remap from="rosout_agg" to="null" />
    </node>

    <node name="segnet" pkg="segnet_tradr" type="imageSemseg.py" output="screen">
        
    </node>
    <node name="classPcl" pkg="object_class_pcl" type="semseg_to_pcl.py" output="screen">

    </node>
    <node name="crf" pkg="object_class_pcl" type="semseg_to_pcl.py" output="screen">

    </node>
    <node name="image_view" pkg="image_view" type="image_view" respawn="false" output="screen">
    <param name="image_transport" value="compressed" />
    <remap from="image" to="$(arg cam_dest)/$(arg cam_used)/image" />
    </node>

    
</launch>
